xmtp_mls/src/storage/encrypted_store/refresh_state.rs

```diff
@@ -83,7 +85,7 @@ impl DbConnection {
                     entity_kind,
                     cursor: 0,
                 };
-                new_state.store(self)?;
+                new_state.store_or_ignore(self)?;
```

Comment:
This fix was a bit of a journey. Streaming got fast enough we uncovered a race condition that has been there the whole time. When you add a member to a group and have a stream running there are two syncs that happen. One from the initial add, and one triggered by reading the message from the stream.

If there is a database error like this in the sync originating from the stream, it won't wait to process the message. Then the stream handler will try to look up the message before it has been saved to the DB and return None, making it seem like the message couldn't be processed when in fact it was just processed later.

All this was causing the `test_subscribe_membership_changes` test to behave flakily.